### PR TITLE
URL for the documental APP priority-alerts

### PR DESCRIPTION
**Pull request by smartsena**

### CODE REVIEW

File: documental/filters.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+class DocumentalDocsFilter(rest_framework.FilterSet):
+    """Django filter `models.DocumentalDocs` data.
+
+    Filter:
+        * acao_id (integer): action identifier to be filtered
+        * co_cr (list): filtering Regional Coordination using code.
+        * co_funai (list): filtering Indigenou Lands using Funai code.
+        * start_date (str): filtering start date.
+        * end_date (str): filteringend end date.
```
Comment (dagnaldo) [inline documental/filters.py:29]: *filtering end date*

File: documental/filters.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+):    
+    """Base class used for creating IN lookup filters to filter characters."""
+    pass
+
+
+class DocumentalDocsFilter(rest_framework.FilterSet):
+    """Django filter `models.DocumentalDocs` data.
+
+    Filter:
+        * acao_id (integer): action identifier to be filtered
```
Comment (dagnaldo) [inline documental/filters.py:25]: **int**, only.

File: documental/filters.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        * end_date (str): filteringend end date.
+    """
+    
+    acao_id = rest_framework.BaseInFilter(
+        field_name='acao_id',
+        lookup_expr='exact',
+        required=True
+    )
+    
+    co_cr = CharInFilter(
```
Comment (dagnaldo) [inline documental/filters.py:38]: Is it a required filter?

File: documental/filters.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        field_name='co_cr',
+        lookup_expr='in'
+    )
+
+    co_funai = NumberInFilter(
```
Comment (dagnaldo) [inline documental/filters.py:43]: Is it a required filter?

File: documental/serializers.py [hunk 5]
```diff
@@ -1,6 +1,7 @@
 from rest_framework import serializers
 
 from documental import models
+from django.conf import settings
```
Comment (dagnaldo) [inline documental/serializers.py:4]: local imports, as `from documental import models` migth be at the end of imports.

File: documental/urls.py [hunk 6]
```diff
@@ -7,5 +7,10 @@
         'list-actions/',
         views.ActionListView.as_view(),
         name='listar-action'
+    ),
+    path(
+        'get-doc/',
```
Comment (dagnaldo) [inline documental/urls.py:12]: what about `list/`?

File: documental/views.py [hunk 7]
```diff
@@ -20,3 +20,17 @@ class ActionListView(generics.ListAPIView):
     serializer_class = serializers.ActionListSerializers
 
 
+class DocumentalDocsViews(generics.ListAPIView):
```
Comment (dagnaldo) [inline documental/views.py:23]: DocumentalListView.

File: documental/serializers.py [hunk 8]
```diff
@@ -14,3 +15,51 @@ class Meta:
...
+
+class MapasUsoOcupacaoSoloSerializers(serializers.ModelSerializer):
+    """Serializer to return datas to action category USO_OCUPAÇÃO_DO_SOLO to
+    `models.DocumentalDocs` data.
+    """
+    usuario_id = UsuarioSerializers()
+    acao_id = ActionListSerializers()
+    class Meta:
+        """Meta class for `MapasUsoOcupacaoSoloSerializers` serializer."""
+        model = models.DocumentalDocs
```
Comment (dagnaldo) [inline documental/serializers.py:38]: missing blank line.

**Review by dagnaldo — CHANGES_REQUESTED** [review]
Take a look of blanklines and and names of variables and classes.

File: documental/filters.py (hunk 4 above)
Comment (smartsena) [reply documental/filters.py:43]: is required = False

File: documental/serializers.py (hunk 5 above)
Comment (smartsena) [reply documental/serializers.py:4]: Removed.

File: documental/serializers.py [hunk 9]
```diff
@@ -14,3 +15,51 @@ class Meta:
         )
 
 
+class UsuarioSerializers(serializers.ModelSerializer):
+    """Serializer to return user who entered the document `models.Usuario` data.
```
[the remark this answers is not in the document]
Comment (smartsena) [reply documental/serializers.py:19]: """Serializer to return the user who entered the `models.Usuario` data."""

File: documental/serializers.py [hunk 10]
```diff
@@ -14,3 +15,51 @@ class Meta:
...
+    """
```
[the remark this answers is not in the document]
Comment (smartsena) [reply documental/serializers.py:20]: Adjusted.

File: documental/serializers.py [hunk 11]
```diff
@@ -14,3 +15,51 @@ class Meta:
...
+        """Meta class for `UsuarioSerializers` serializer."""
+        model = models.Usuario
+        fields = (
+            'id',
+            'first_name',
+        )
+
+
+class MapasUsoOcupacaoSoloSerializers(serializers.ModelSerializer):
+    """Serializer to return datas to action category USO_OCUPAÇÃO_DO_SOLO to
```
[the remark this answers is not in the document]
Comment (smartsena) [reply documental/serializers.py:31]: Adjusted.

File: documental/serializers.py [hunk 12]
```diff
@@ -14,3 +15,51 @@ class Meta:
...
+    `models.DocumentalDocs` data.
+    """
+    usuario_id = UsuarioSerializers()
```
[the remark this answers is not in the document]
Comment (smartsena) [reply documental/serializers.py:34]: ok

File: documental/serializers.py [hunk 13]
```diff
@@ -14,3 +15,51 @@ class Meta:
...
+    acao_id = ActionListSerializers()
+    class Meta:
```
[the remark this answers is not in the document]
Comment (smartsena) [reply documental/serializers.py:36]: ok

File: documental/urls.py (hunk 6 above)
Comment (smartsena) [reply documental/urls.py:12]: Adjusted

File: documental/views.py [hunk 14]
```diff
@@ -20,3 +20,17 @@ class ActionListView(generics.ListAPIView):
...
+    """Returns `models.DocumentosDoc` data acoording to the selected actions.
+    Filter:
+        * acao_id (integer) (mandatory): action identifier to be filtered
+        * co_cr (list): filtering Regional Coordination using code.
+        * co_funai (list): filtering Indigenou Lands using Funai code.
+        * start_date (str): filtering start date.
+        * end_date (str): filteringend end date.
```
[the remark this answers is not in the document]
Comment (smartsena) [reply documental/views.py:30]: Adjusted.

File: documental/serializers.py [hunk 15]
```diff
@@ -14,3 +15,51 @@ class Meta:
...
+    
+    def to_representation(self, instance):
+        """Method to return in `DocumentosTISerializers` the full URL to
+        download the documents in `models.DocumentalDocs`  
+                
+        Returns:
+            url_document -> str
+        """
+        url_document = super().to_representation(instance)
+        url_document['url_doc'] = settings.DOCUMENTOS + instance.path_documento
```
[the remark this answers is not in the document]
Comment (smartsena) [reply documental/serializers.py:62]: Added urljoin in this function.

File: documental/views.py (hunk 7 above)
Comment (smartsena) [reply documental/views.py:23]: Renamed class in views.py

File: documental/urls.py [hunk 16]
```diff
@@ -7,5 +7,10 @@
...
+        views.DocumentalDocsViews.as_view(),
```
[the remark this answers is not in the document]
Comment (smartsena) [reply documental/urls.py:13]: Renamed url path to 'list/'

File: support/fixtures/support.yaml [hunk 17]
```diff
@@ -133,7 +133,7 @@
 - model: support.layersgroup
   pk: 19
   fields:
-      name: Monitoramento Diário
+      name: Monitoramento Antropismo
```
Comment (dagnaldo) [inline support/fixtures/support.yaml:136]: Antropismo Consolidado.

File: documental/views.py [hunk 18]
```diff
@@ -1,22 +1,42 @@
...
+        * co_cr (list): filtering Regional Coordination using code.
+        * co_funai (list): filtering Indigenou Lands using Funai code.
+        * start_date (str): filtering start date.
+        * end_date (str): filtering end date.
+    """
+
+    queryset = models.DocumentalDocs.objects.all().order_by('dt_cadastro')
+    serializer_class = serializers.MapasUsoOcupacaoSoloSerializers
+    filterset_class = documental_filters.DocumentalDocsFilter
+    filter_backends = DjangoFilterBackend,
```
Comment (dagnaldo) [inline documental/views.py:41]: Change to `filter_backends =(DjangoFilterBackend, )`

File: documental/serializers.py [hunk 19]
```diff
@@ -14,3 +18,55 @@ class Meta:
...
+            'acao_id',
+            'usuario_id',
+        )
+    
+    def to_representation(self, instance):
+        """Method to return in `DocumentosTISerializers` the full URL to
+        download the documents in `models.DocumentalDocs`  
+                
+        Returns:
+            url_document -> str
```
Comment (dagnaldo) [inline documental/serializers.py:65]: Returns might be:
```
Returns:
    str: url to document.
```

**Review by dagnaldo — CHANGES_REQUESTED** [review]

File: support/fixtures/support.yaml (hunk 17 above)
Comment (smartsena) [reply support/fixtures/support.yaml:136]: ok

File: documental/views.py (hunk 18 above)
Comment (smartsena) [reply documental/views.py:41]: ok

File: documental/serializers.py (hunk 19 above)
Comment (smartsena) [reply documental/serializers.py:65]: ok

File: documental/serializers.py [hunk 20]
```diff
@@ -14,3 +18,55 @@ class Meta:
...
+    def to_representation(self, instance):
+        """Method to return in `DocumentosTISerializers` the full URL to
+        download the documents in `models.DocumentalDocs`  
+                
+        Returns:
+            str: url to document
+        """
+
+        url_document = super().to_representation(instance)
+        url_document['url_doc'] = urllib.parse.urljoin (settings.DOCUMENTOS, instance.path_documento)
```
Comment (dagnaldo) [inline documental/serializers.py:69]: Remove space between `urljoin` and `(`

Might be like this: 
`urllib.parse.urljoin (settings.DOCUMENTOS, instance.path_documento)`

**Review by dagnaldo — CHANGES_REQUESTED** [review]

**Review by dagnaldo — APPROVED** [review]

**Pull request merged**